[minizip] build failure arm-neon-android

### Operating system

Linux

### Compiler

Android NDK v26

### Steps to reproduce the behavior

```Shell
./vcpkg install minizip:arm-neon-android
```


### Failure logs

```
Change Dir: '/home/ankurv/devel/vcpkg.test/buildtrees/minizip/arm-neon-android-dbg'

Run Build Command(s): /usr/bin/ninja -v -v -j25 install
[1/6] /home/ankurv/devel/bin/android/sdk/ndk/26.0.10792818/toolchains/llvm/prebuilt/linux-x86_64/bin/clang --target=armv7-none-linux-androideabi21 --sysroot=/home/ankurv/devel/bin/android/sdk/ndk/26.0.10792818/toolchains/llvm/prebuilt/linux-x86_64/sysroot -DIOAPI_NO_64 -D_ZLIB_H -isystem /home/ankurv/devel/vcpkg.test/installed/arm-neon-android/include -g -DANDROID -fdata-sections -ffunction-sections -funwind-tables -fstack-protector-strong -no-canonical-prefixes -D_FORTIFY_SOURCE=2 -march=armv7-a -mthumb -Wformat -Werror=format-security  -fPIC   -fno-limit-debug-info    -fPIC -MD -MT CMakeFiles/minizip.dir/ioapi.c.o -MF CMakeFiles/minizip.dir/ioapi.c.o.d -o CMakeFiles/minizip.dir/ioapi.c.o -c /home/ankurv/devel/vcpkg.test/buildtrees/minizip/src/v1.3-a9caa60c28.clean/contrib/minizip/ioapi.c
FAILED: CMakeFiles/minizip.dir/ioapi.c.o 
/home/ankurv/devel/bin/android/sdk/ndk/26.0.10792818/toolchains/llvm/prebuilt/linux-x86_64/bin/clang --target=armv7-none-linux-androideabi21 --sysroot=/home/ankurv/devel/bin/android/sdk/ndk/26.0.10792818/toolchains/llvm/prebuilt/linux-x86_64/sysroot -DIOAPI_NO_64 -D_ZLIB_H -isystem /home/ankurv/devel/vcpkg.test/installed/arm-neon-android/include -g -DANDROID -fdata-sections -ffunction-sections -funwind-tables -fstack-protector-strong -no-canonical-prefixes -D_FORTIFY_SOURCE=2 -march=armv7-a -mthumb -Wformat -Werror=format-security  -fPIC   -fno-limit-debug-info    -fPIC -MD -MT CMakeFiles/minizip.dir/ioapi.c.o -MF CMakeFiles/minizip.dir/ioapi.c.o.d -o CMakeFiles/minizip.dir/ioapi.c.o -c /home/ankurv/devel/vcpkg.test/buildtrees/minizip/src/v1.3-a9caa60c28.clean/contrib/minizip/ioapi.c
/home/ankurv/devel/vcpkg.test/buildtrees/minizip/src/v1.3-a9caa60c28.clean/contrib/minizip/ioapi.c:144:21: error: call to undeclared function 'ftello'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
    ret = (ZPOS64_T)FTELLO_FUNC((FILE *)stream);
                    ^
/home/ankurv/devel/vcpkg.test/buildtrees/minizip/src/v1.3-a9caa60c28.clean/contrib/minizip/ioapi.c:20:29: note: expanded from macro 'FTELLO_FUNC'
#define FTELLO_FUNC(stream) ftello(stream)
                            ^
/home/ankurv/devel/vcpkg.test/buildtrees/minizip/src/v1.3-a9caa60c28.clean/contrib/minizip/ioapi.c:144:21: note: did you mean 'ftell'?
/home/ankurv/devel/vcpkg.test/buildtrees/minizip/src/v1.3-a9caa60c28.clean/contrib/minizip/ioapi.c:20:29: note: expanded from macro 'FTELLO_FUNC'
#define FTELLO_FUNC(stream) ftello(stream)
                            ^
/home/ankurv/devel/bin/android/sdk/ndk/26.0.10792818/toolchains/llvm/prebuilt/linux-x86_64/sysroot/usr/include/stdio.h:216:6: note: 'ftell' declared here
long ftell(FILE* _Nonnull __fp);
     ^
/home/ankurv/devel/vcpkg.test/buildtrees/minizip/src/v1.3-a9caa60c28.clean/contrib/minizip/ioapi.c:190:8: error: call to undeclared function 'fseeko'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
    if(FSEEKO_FUNC((FILE *)stream, (z_off64_t)offset, fseek_origin) != 0)
       ^
/home/ankurv/devel/vcpkg.test/buildtrees/minizip/src/v1.3-a9caa60c28.clean/contrib/minizip/ioapi.c:21:45: note: expanded from macro 'FSEEKO_FUNC'
#define FSEEKO_FUNC(stream, offset, origin) fseeko(stream, offset, origin)
                                            ^
/home/ankurv/devel/vcpkg.test/buildtrees/minizip/src/v1.3-a9caa60c28.clean/contrib/minizip/ioapi.c:190:8: note: did you mean 'fseek'?
/home/ankurv/devel/vcpkg.test/buildtrees/minizip/src/v1.3-a9caa60c28.clean/contrib/minizip/ioapi.c:21:45: note: expanded from macro 'FSEEKO_FUNC'
#define FSEEKO_FUNC(stream, offset, origin) fseeko(stream, offset, origin)
                                            ^
/home/ankurv/devel/bin/android/sdk/ndk/26.0.10792818/toolchains/llvm/prebuilt/linux-x86_64/sysroot/usr/include/stdio.h:215:5: note: 'fseek' declared here
int fseek(FILE* _Nonnull __fp, long __offset, int __whence);
    ^
2 errors generated.
[2/6] /home/ankurv/devel/bin/android/sdk/ndk/26.0.10792818/toolchains/llvm/prebuilt/linux-x86_64/bin/clang --target=armv7-none-linux-androideabi21 --sysroot=/home/ankurv/devel/bin/android/sdk/ndk/26.0.10792818/toolchains/llvm/prebuilt/linux-x86_64/sysroot -DIOAPI_NO_64 -D_ZLIB_H -isystem /home/ankurv/devel/vcpkg.test/installed/arm-neon-android/include -g -DANDROID -fdata-sections -ffunction-sections -funwind-tables -fstack-protector-strong -no-canonical-prefixes -D_FORTIFY_SOURCE=2 -march=armv7-a -mthumb -Wformat -Werror=format-security  -fPIC   -fno-limit-debug-info    -fPIC -MD -MT CMakeFiles/minizip.dir/mztools.c.o -MF CMakeFiles/minizip.dir/mztools.c.o.d -o CMakeFiles/minizip.dir/mztools.c.o -c /home/ankurv/devel/vcpkg.test/buildtrees/minizip/src/v1.3-a9caa60c28.clean/contrib/minizip/mztools.c
[3/6] /home/ankurv/devel/bin/android/sdk/ndk/26.0.10792818/toolchains/llvm/prebuilt/linux-x86_64/bin/clang --target=armv7-none-linux-androideabi21 --sysroot=/home/ankurv/devel/bin/android/sdk/ndk/26.0.10792818/toolchains/llvm/prebuilt/linux-x86_64/sysroot -DIOAPI_NO_64 -D_ZLIB_H -isystem /home/ankurv/devel/vcpkg.test/installed/arm-neon-android/include -g -DANDROID -fdata-sections -ffunction-sections -funwind-tables -fstack-protector-strong -no-canonical-prefixes -D_FORTIFY_SOURCE=2 -march=armv7-a -mthumb -Wformat -Werror=format-security  -fPIC   -fno-limit-debug-info    -fPIC -MD -MT CMakeFiles/minizip.dir/unzip.c.o -MF CMakeFiles/minizip.dir/unzip.c.o.d -o CMakeFiles/minizip.dir/unzip.c.o -c /home/ankurv/devel/vcpkg.test/buildtrees/minizip/src/v1.3-a9caa60c28.clean/contrib/minizip/unzip.c
[4/6] /home/ankurv/devel/bin/android/sdk/ndk/26.0.10792818/toolchains/llvm/prebuilt/linux-x86_64/bin/clang --target=armv7-none-linux-androideabi21 --sysroot=/home/ankurv/devel/bin/android/sdk/ndk/26.0.10792818/toolchains/llvm/prebuilt/linux-x86_64/sysroot -DIOAPI_NO_64 -D_ZLIB_H -isystem /home/ankurv/devel/vcpkg.test/installed/arm-neon-android/include -g -DANDROID -fdata-sections -ffunction-sections -funwind-tables -fstack-protector-strong -no-canonical-prefixes -D_FORTIFY_SOURCE=2 -march=armv7-a -mthumb -Wformat -Werror=format-security  -fPIC   -fno-limit-debug-info    -fPIC -MD -MT CMakeFiles/minizip.dir/zip.c.o -MF CMakeFiles/minizip.dir/zip.c.o.d -o CMakeFiles/minizip.dir/zip.c.o -c /home/ankurv/devel/vcpkg.test/buildtrees/minizip/src/v1.3-a9caa60c28.clean/contrib/minizip/zip.c
ninja: build stopped: subcommand failed.

```

### Additional context

_No response_

--- a/ports/minizip/CMakeLists.txt
+++ b/ports/minizip/CMakeLists.txt
@@ -41,10 +41,6 @@ else()
   message(STATUS "Building without bzip2 support")
 endif()
 
-if(ANDROID)
-  target_compile_definitions(minizip PRIVATE IOAPI_NO_64)
-endif()
-
 if(CMAKE_SYSTEM_NAME STREQUAL "WindowsStore")
   target_compile_definitions(minizip PRIVATE -DIOWIN32_USING_WINRT_API)
 endif()

--- a/ports/minizip/android-fileapi.patch
+++ b/ports/minizip/android-fileapi.patch
@@ -1,0 +1,17 @@
+diff --git a/contrib/minizip/ioapi.h b/contrib/minizip/ioapi.h
+index c588a18..b5395e2 100644
+--- a/contrib/minizip/ioapi.h
++++ b/contrib/minizip/ioapi.h
+@@ -21,6 +21,12 @@
+ #ifndef _ZLIBIOAPI64_H
+ #define _ZLIBIOAPI64_H
+ 
++#if defined(__ANDROID_API__) && __ANDROID_API__ < 24
++    // Cf. https://android.googlesource.com/platform/bionic/+/master/docs/32-bit-abi.md#32_bit-and
++    // stdio functions for off_t are incomplete.
++    #define USE_FILE32API
++#endif
++
+ #if (!defined(_WIN32)) && (!defined(WIN32)) && (!defined(__APPLE__))
+ 
+   // Linux needs this to support file operation on files larger then 4+GB

--- a/ports/minizip/portfile.cmake
+++ b/ports/minizip/portfile.cmake
@@ -12,13 +12,12 @@ vcpkg_from_github(
         android-fileapi.patch
 )
 
-vcpkg_cmake_get_vars(cmake_vars_file)
-include("${cmake_vars_file}")
-
 # Maintainer switch: Temporarily set this to 1 to re-generate the lists
 # of exported symbols. This is needed when the version is bumped.
 set(GENERATE_SYMBOLS 0)
 if(GENERATE_SYMBOLS)
+    vcpkg_cmake_get_vars(cmake_vars_file)
+    include("${cmake_vars_file}")
     if(VCPKG_DETECTED_CMAKE_C_COMPILER_ID STREQUAL "MSVC")
         vcpkg_check_linkage(ONLY_STATIC_LIBRARY)
     else()

--- a/ports/minizip/portfile.cmake
+++ b/ports/minizip/portfile.cmake
@@ -2,13 +2,14 @@
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO madler/zlib
-    REF v${VERSION}
+    REF "v${VERSION}"
     SHA512 78eecf335b14af1f7188c039a4d5297b74464d61156e4f12a485c74beec7d62c4159584ad482a07ec57ae2616d58873e45b09cb8ea822bb5b17e43d163df84e9
     HEAD_REF master
     PATCHES
         0001-remove-ifndef-NOUNCRYPT.patch
         0002-add-declaration-for-mkdir.patch
         pkgconfig.patch
+        android-fileapi.patch
 )
 
 vcpkg_cmake_get_vars(cmake_vars_file)

--- a/ports/minizip/vcpkg.json
+++ b/ports/minizip/vcpkg.json
@@ -1,6 +1,7 @@
 {
   "name": "minizip",
   "version": "1.3",
+  "port-version": 1,
   "description": "Minizip zip file manipulation library",
   "homepage": "https://github.com/madler/zlib",
   "license": "Zlib",

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -5454,7 +5454,7 @@
     },
     "minizip": {
       "baseline": "1.3",
-      "port-version": 0
+      "port-version": 1
     },
     "minizip-ng": {
       "baseline": "4.0.0",

--- a/versions/m-/minizip.json
+++ b/versions/m-/minizip.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "91fe1aaa8a5d696bda657a3d2b687fecfa92a7c3",
+      "version": "1.3",
+      "port-version": 1
+    },
+    {
       "git-tree": "2295a5b26e2cbbbf59706f08dc1c5717da951bd1",
       "version": "1.3",
       "port-version": 0

